### PR TITLE
test(events): add stress test suite for ingestion pipeline . Closes #1531

### LINKED ISSUE #1531: Add high-volume stress and integration test suite for event ingestion pipeline
The current unit test suite   in develop branch for event ingestion verifies basic functional paths, but lacks explicit boundary, stress, and concurrency checks to ensure system stability under high-load conditions in production.

To ensure pipeline resilience as honeypot telemetry scales, we need comprehensive stress tests covering batch limits, concurrent sensor submissions, database transaction atomicity, and array deduplication.

### PR DESCRIPTION
**Pull request by drona-gyawali**
# Description

This PR introduces a comprehensive stress test suite for event ingestion pipeline 

While standard unit tests cover nominal execution paths, this suite explicitly validates system resilience under high-volume conditions, concurrency, edge-case array payloads, and mid-pipeline worker failures.

### Related issues
closes : #1531 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Chore (refactoring, dependency updates, CI/CD changes, code cleanup, docs-only changes).

# Checklist

Please complete this checklist carefully. It helps guide your contribution and lets maintainers verify that all requirements are met.

### Formalities

- [x] I have read and understood the rules about [how to Contribute](https://github.com/GreedyBear-Project/GreedyBear/wiki/Contribute) to this project.
- [x] I chose an appropriate title for the pull request in the form: `<feature name>. Closes #999`
- [x] My branch is based on `develop`.
- [x] The pull request is for the branch `develop`.
- [x] I have reviewed and verified any LLM-generated code included in this PR.

### Docs and tests

- [x] I documented my code changes with docstrings and/or comments.
- [ ] I have checked if my changes affect user-facing behavior that is described in the [docs](https://github.com/GreedyBear-Project/GreedyBear/wiki). If so, I also included an update to the [wiki](https://github.com/GreedyBear-Project/GreedyBear/wiki) in the description of this PR.
- [x] Linter (`Ruff`) gave 0 errors. If you have correctly installed [pre-commit](https://github.com/GreedyBear-Project/GreedyBear/wiki/Contribute), it does these checks and adjustments on your behalf.
- [x] I have added tests for the feature/bug I solved.
- [x] All the tests gave 0 errors.

### GUI changes

Ignore this section if you did not make any changes to the GUI.

- [ ] I have provided a screenshot of the result in the PR.
- [ ] I have created new frontend tests for the new component or updated existing ones.
  
# Review process

- We encourage you to create a draft PR first, even when your changes are incomplete. This way you refine your code while we can track your progress and actively review and help.
- If you think your draft PR is ready to be reviewed by the maintainers, click the corresponding button. Your draft PR will become a real PR.
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem.
- Every time you make changes to the PR and you think the work is done, you should explicitly ask for a review. After receiving a "change request", address the feedback and click "request re-review" next to the reviewer's profile picture at the top right.

### CODE REVIEW

File: tests/test_event_stress.py [hunk 1]
```diff
@@ -0,0 +1,752 @@
...
+    return base
+
+
+def make_batch(api_source, task_id="stress-batch-1", batch_status="pending"):
+    return EventStatus.objects.create(
+        api_source=api_source,
+        task_id=task_id,
+        status=batch_status,
+    )
+
```
Comment (regulartim) [inline tests/test_event_stress.py]: Is this dead code?

File: tests/test_event_stress.py [hunk 2]
```diff
@@ -0,0 +1,752 @@
...
+                related_url=f"http://evil.com/malware{i}",
+            )
+            for i in range(5_000)
+        ]
+        res = self.client.post(EVENTS_URL, {"events": events}, format="json")
+
+        self.assertEqual(res.status_code, status.HTTP_202_ACCEPTED)
+        task_id = res.data["task_id"]
+        self.assertEqual(RawEvent.objects.filter(batch__task_id=task_id).count(), 5_000)
+
```
Comment (regulartim) [inline tests/test_event_stress.py]: This test overstates what it actually does. It never checks if optional fields are all persisted correctly.

File: tests/test_event_stress.py [hunk 3]
```diff
@@ -0,0 +1,752 @@
...
+class TestProcessCredentialsStress(CustomTestCase):
+    """
+    Push _process_credentials with hundreds of distinct and duplicate
+    credentials to verify no silent truncation, no duplicate DB rows,
+    and correct M2M linkage at scale.
+    """
+
+    def setUp(self):
+        Credential.objects.all().delete()
```
Comment (regulartim) [inline tests/test_event_stress.py]: `.delete()` calls in `setUp` should be no-ops. The transaction rollback already does that. Can you check? Also for your other subclasses of `CustomTestCase`.

**Review by regulartim — CHANGES_REQUESTED** [review]
Hey @drona-gyawali ! Looks good, thank you! 

In a perfect world, the tests would (more often) assert a bound that a regression can cross. Like how may queries were executed. You used `assertNumQueries` two times, but if would also be interesting to test the number of queries during the actual event creation.

File: tests/test_event_stress.py (hunk 1 above)
Comment (drona-gyawali) [reply tests/test_event_stress.py]: yes, make_batch as an instance method on TestProcessIncomingEventStress was doing that job. I will remove that

File: tests/test_event_stress.py (hunk 3 above)
Comment (drona-gyawali) [reply tests/test_event_stress.py]: confirmed. CustomTestCase wraps each test in a transaction that rolls back after the test completes, so the table is already empty at setUp time, So i will remove them. And I also found  that removing .delete() calls revealed  isolation bug(several CommandSequence tests were using .first() and table-wide .count())  which picked up rows from other tests that had run earlier , so i will update code by  filtering on commands_hash (the unique constraint) so each test only sees the rows it created.

**Review by regulartim — APPROVED** [review]
Nice, thanks! :)
